support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -248,9 +265,9 @@ object DigitalSubscriptionGiftRedemption {
         HandlerResult(SendThankYouEmailState(
           state.requestId,
           state.user,
-          state.giftRecipient,
+          None,
```

Comment:
there's never a gift recipient for a redemption actually.

I was having a think about approaches for the State.. some thoughts below that we can think about collectively:

The one size fits all model makes sense if most execution types are similar, but now that we have more disparate things to do, some of these are starting to make less sense in some situations than others.  I can see this on line 275 just below here:
![image](https://user-images.githubusercontent.com/7304387/95897185-07f15500-0d85-11eb-8c65-ca65e1e82c9e.png)

which we could indeed add more options, but maybe they should not exist at all if it's a redemption (ie use a sum type rather than a product type)
I have slightly pushed the sum type approach with recent PRs, making the GiftRecipient type GWGift+DSGift, 

Downside: does complicate the serialisers a bit.  And makes the raw JSON diverge a bit from the scala representation.
Upside: But it does give us the type safety and lack of having a lot of Options or empty strings to fill in.  And also means we don't have to keep getting things out of Options if they "must" exist in reality.

Reply:
Happy to discuss this further offline

Reply:
conveniently, next in the list for post standup pairing time is you and me!